Ports slide projectors from nebula

CrimsonShrike opened the pull request:
🆑CrimsonShrike
rscadd: Slide projectors. Maps one to the briefing room on Deck 1
/🆑

Amaze your colleagues with your scientific discoveries! Share forgettable memos! Bore everyone with your vacation pictures!

All of this and more with the slide projector!

![image](https://user-images.githubusercontent.com/29737699/111806065-82407b00-88c9-11eb-848b-9a05fba87b11.png)

PsiOmegaDelta commented on code/game/objects/items/devices/slide_projector.dm:
```diff
@@ -0,0 +1,207 @@
+	var/obj/effect/projection/projection
+
+/obj/item/storage/slide_projector/Destroy()
+	current_slide = null
```
Any inserted slides just disappear into null space instead of being either dropped or deleted?

PsiOmegaDelta replied:
Looks like slides actually remain "external"
How does the projector handle any referenced slides being deleted?

CrimsonShrike replied:
Poorly, I assume

CrimsonShrike replied:
Will take a look. Listening to destruction is an option but may be able to leverage whatever it is storage does.

CrimsonShrike replied:
Hijacked storage logic. On predeletion it will be nulled

PsiOmegaDelta commented on code/game/objects/items/devices/slide_projector.dm:
```diff
@@ -0,0 +1,207 @@
+	max_w_class = ITEM_SIZE_SMALL
+	max_storage_space = BASE_STORAGE_CAPACITY(ITEM_SIZE_SMALL)
+	use_sound = 'sound/effects/storage/toolbox.ogg'
+	var/list/global/projection_types = list(
```
`static` is the more typical convention for type-shared vars

SierraKomodo commented on code/game/objects/items/devices/slide_projector.dm:
```diff
@@ -0,0 +1,211 @@
+		set_slide(null)
+	
+/obj/item/storage/slide_projector/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!current_slide)
+		return
```
User feedback.

```suggestion
	if(!current_slide)
		to_chat(user, SPAN_WARNING("\The [src] does not have a slide loaded."))
		return
```

SierraKomodo commented on code/game/objects/items/devices/slide_projector.dm:
```diff
@@ -0,0 +1,211 @@
+	else
+		data += "Projector inactive"
+	
+	var/table = list("<table><th>#<th>SLIDE<th>SHOW")
+	var/i = 1
+	for(var/obj/item/I in contents)
+		table += "<tr><td>#[i]"
+		if(I == current_slide)
+			table += "<td><b>[I.name]</b><td>SHOWING"
+		else
+			table += "<td>[I.name]<td><a href='?src=\ref[src];set_active=[i]'>SHOW</a>"
+		i++
```
HTML closing tags

```suggestion
	var/table = list("<table><tr><th>#</th><th>SLIDE</th><th>SHOW</th></tr>")
	var/i = 1
	for(var/obj/item/I in contents)
		table += "<tr><td>#[i]</td>"
		if(I == current_slide)
			table += "<td><b>[I.name]</b></td><td>SHOWING</td>"
		else
			table += "<td>[I.name]</td><td><a href='?src=\ref[src];set_active=[i]'>SHOW</a></td>"
		table += "</tr>"
		i++
	table += "</table>"
```

CrimsonShrike committed:
Ports slide projectors from nebula


Update code/game/objects/items/devices/slide_projector.dm

Co-authored-by: Sierra Brown (SierraKomodo) <SierraKomodo@users.noreply.github.com>
Update code/game/objects/items/devices/slide_projector.dm

Co-authored-by: Sierra Brown (SierraKomodo) <SierraKomodo@users.noreply.github.com>

SierraKomodo reviewed (commented):
Is being able to put items into the projector and project them intentional?

CrimsonShrike commented:
Ye, that's why they use base projection type, with photos and writings using alternate types
